test/expression_test.ts

```diff
@@ -92,4 +92,59 @@ describe('expressions', () => {
     expectTranslate('foo.bar;').to.equal('foo.bar;');
     expectTranslate('foo[bar];').to.equal('foo[bar];');
   });
+  it('should cast numeric values to int when bitwise operators are used for a LHS exp RHS numeric expression',
+     () => {
+       expectTranslate(`var z: number = x | y`).to.equal("num z = (x as int) | (y as int);");
+
+       expectTranslate(`var z: number = x & y`).to.equal("num z = (x as int) & (y as int);");
+
+       expectTranslate(`var z: number = x ^ y`).to.equal("num z = (x as int) ^ (y as int);");
+
+       expectTranslate(`var z: number = x << y`).to.equal("num z = (x as int) << (y as int);");
+
+       expectTranslate(`var z: number = x >> y`).to.equal("num z = (x as int) >> (y as int);");
+     });
+  it('should cast numeric values to int when a shorthand bitwise assignment expression is used',
+     () => {
+       expectTranslate('var y: number = (y |= x)')
+           .to.equal('num y = (y = (y as int) | (x as int));');
+
+       expectTranslate('var y: number = (y &= x)')
+           .to.equal('num y = (y = (y as int) & (x as int));');
+
+       expectTranslate('var y: number = (y ^= x)')
+           .to.equal('num y = (y = (y as int) ^ (x as int));');
+
+       expectTranslate('var y: number = (y <<= x)')
+           .to.equal('num y = (y = (y as int) << (x as int));');
+
+       expectTranslate('var y: number = (y >>= x)')
+           .to.equal('num y = (y = (y as int) >> (x as int));');
+     });
+  it('should cast numeric values to int when bitwise data is evaluated as an input param into a function',
```

Comment:
I don't think this test adds value - it's just the regular operation of ts2dart, unrelated to your change. Please drop it.